pkg/controllers/networkpolicy/policy_controller.go

```diff
@@ -232,14 +188,19 @@ func (c *PolicyController) processNextItem() bool {
 	return true
 }
 
-// syncToCalico syncs the given update to the Calico datastore.
-func (c *PolicyController) syncToCalico(key string) error {
+// syncToDatastore syncs the given update to the Calico datastore. The provided key can be used to
+// find the corresponding resource within the resource cache. If the resource for the provided key
+// exists in the cache, then the value should be written to the datastore. If it does not exist
+// in the cache, then it should be deleted from the datastore.
+func (c *policyController) syncToDatastore(key string) error {
 	// Check if it exists in the controller's cache.
-	obj, exists := c.calicoObjCache.Get(key)
+	obj, exists := c.resourceCache.Get(key)
 	if !exists {
 		// The object no longer exists - delete from the datastore.
-		log.Infof("Deleting Policy %s from Calico datastore", key)
-		if err := c.calicoClient.Policies().Delete(api.PolicyMetadata{Name: key}); err != nil {
+		log.Infof("Deleting NetworkPolicy %s from Calico datastore", key)
+		ns, name := converter.NewPolicyConverter().DeleteArgsFromKey(key)
+		_, err := c.calicoClient.NetworkPolicies().Delete(context.Background(), ns, name, options.DeleteOptions{})
+		if err != nil {
 			if _, ok := err.(errors.ErrorResourceDoesNotExist); !ok {
```

Comment:
I think this can be simplified as:
```go
_, err := c.calicoClient.NetworkPolicies().Delete(context.Background(), ns, name, options.DeleteOptions{})
if _, ok :=  err.(errors.ErrorResourceDoesNotExist); !ok {
    return err
}
return nil
```